assets/javascripts/discourse/components/topic-chat-float.js

```diff
@@ -347,12 +347,14 @@ export default Component.extend({
 
   @action
   switchChannel(channel) {
-    if (this.site.mobileView) {
-      return this.router.transitionTo(
-        "chat.channel",
-        channel.id,
-        channel.title
-      );
+    if (this.site.mobileView || this.chat.onChatPage()) {
+      next(() => {
```

Comment:
Why `next()`?

Reply:
Mentioned above - chat service marks channel as read, and this avoids a race condition so that the channel will never look like it's in "preview" mode